StortSpelProjekt/Game/src/Gamefiles/Components/AiComponent.cpp

```diff
@@ -568,6 +612,12 @@ void component::AiComponent::updateRange(double dt)
 			{
 				if (m_IntervalTimeAccumulator >= m_AttackInterval && m_DistanceToPlayer >= m_AttackingDistance / 8.0f)
 				{
+					if (m_ShouldPlayAttackAnimation)
```

Comment:
After testing, I found that the actual shot happens noticeably after the enemy moves his harm. I would suggest checking in this if-statement whether m_SpeedTimeAccumulator is larger than something like m_AttackSpeed / 2. You'll have to test and see which number works best, but something like that should give a more desirable result